cxplat/src/tags.h

```diff
@@ -0,0 +1,9 @@
+// Copyright (c) Microsoft Corporation
+// SPDX-License-Identifier: MIT
+#pragma once
+
+// Pool tags used by the cxplat library.
+#define CXPLAT_DEFAULT_TAG 'edxc'
+#define CXPLAT_PREEMPTIBLE_WORK_ITEM_TAG 'wpxc'
+#define CXPLAT_STRING_TAG 'tsxc'
+#define CXPLAT_UTF8_STRING_TAG 'suxc'
```

Comment:
Instead of trying to come up with a pretty/nice two-character description for each specific type of tag, QUIC goes a different route:
```c
#define QUIC_POOL_GENERIC                   'CIUQ' // QUIC - Generic QUIC
#define QUIC_POOL_SILO                      '00cQ' // Qc00 - QUIC Silo
#define QUIC_POOL_CONN                      '10cQ' // Qc01 - QUIC connection
#define QUIC_POOL_TP                        '20cQ' // Qc02 - QUIC connection transport parameters
#define QUIC_POOL_STREAM                    '30cQ' // Qc03 - QUIC stream
#define QUIC_POOL_SBUF                      '40cQ' // Qc04 - QUIC stream buffer
#define QUIC_POOL_META                      '50cQ' // Qc05 - QUIC sent frame metadata
#define QUIC_POOL_DATA                      '60cQ' // Qc06 - QUIC datagram buffer
#define QUIC_POOL_TEST                      '70cQ' // Qc07 - QUIC test code
#define QUIC_POOL_PERF                      '80cQ' // Qc08 - QUIC perf code
#define QUIC_POOL_TOOL                      '90cQ' // Qc09 - QUIC tool code
#define QUIC_POOL_WORKER                    'A0cQ' // Qc0A - QUIC Worker
```
It's made it much easier to just add a new one.

Reply:
It makes it slower to debug, and since debugging is more common than adding a new tag, I think it's better to optimize for debugging.

Reply:
I'd argue there is no difference. These are my expected steps if there is ever a leak for a given tag `X`:

1. CTRL+F for `X` in the code base to find the `#define` name of the tag, `Y`.
2. CTRL+F for `Y` in the code base to find all usages of that tag.

For these steps, the values of `X` and `Y` don't matter for speed. And I'd argue that `suxc`/`cxus` is not easy enough to remember to be able to directly skip step 1 above, especially if you didn't write the code originally (i.e. new devs on the team).

Reply:
I'd argue that since it makes no difference to you, it should be fine the way it is.  But for folks who are familiar with the code, when I look at things like how much memory is allocated by various tags, it's easier to remember what is what without having to do a code search.

Reply:
So long as you're fine when we merge all the msquic specific cxplat stuff, which uses tons of tags, that we just use our pattern of increasing numbers, I don't care that much what is used for everything else. I still think it's easier to use numbers, but I am fine with this for now.

Reply:
Yes I'm fine with that, but any quic-specific tags wouldn't be in the cxplat project, it'd be in a dependent project.  Just like usersim has its own usersim-specific tags.  I see each consuming project as setting their own tag convention.